### PR TITLE
Wdi10 update

### PR DESCRIPTION
**Pull request by amaseda**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -80,244 +81,476 @@ As you're reviewing the app, try to fill in the blanks in the below Rails-to-Exp
...
 
+### Note
+
+You may notice that we do things a bit differently in this example than we have in previous classes. For example, we use promises instead of callbacks when making Mongoose queries.
+
+This is to demonstrate that there is no single way of creating a Mongoose-Express app. If any of these differences seem confusing, don't hesitate to ask an instructor about them!
+
 ## Before We Continue!
 
 You are welcome to code along during the "I Do's" and "We Do's" in this lesson plan. We do ask, however, that **if you fall behind, do not attempt to catch up during those sections**. Instead, tilt down your screen and watch / take notes.
```
Comment (jshawl) [inline readme.md:94]: maybe remember to screencast? -e.g. slack reminder

File: readme.md [hunk 2]
```diff
@@ -80,244 +81,476 @@ As you're reviewing the app, try to fill in the blanks in the below Rails-to-Exp
...
+In order for us to use Mongoose to communicate with our database, we need to link it up to our Express application. We'll do this by...
+* Establishing a connection with a Mongo database.
+* Define a Mongoose schema and model.
+* Generate some seed data.
 
-> [1b8ff4f](https://www.github.com/ga-wdi-exercises/whenpresident/commit/1b8ff4f)
+### Steps
 
-#### Questions
+1. Install Mongoose via the command line: `npm install --save mongoose`.
```
Comment (jshawl) [inline readme.md:113]: remember to `$ mongod` what does the error look like if you run connection.js, and the server isn't running?

File: readme.md [hunk 3]
```diff
@@ -80,244 +81,476 @@ As you're reviewing the app, try to fill in the blanks in the below Rails-to-Exp
...
+  3. Write Mongoose queries that accomplish the following...
+    - Clears the database of all data, and `.then`...
+    - Inserts our seed data into the database, and `.then`...
+    - Calls `process.exit()`.
+
+We can test this by...
+  1. Running `$ node db/seed.js` in the Terminal.
+  2. Then run `$ mongo` in the Terminal and enter the following commands via the Mongo CLI interface...
+    ```mongo
+    > use whencandidate
```
Comment (jshawl) [inline readme.md]: should this be whenpresident?

File: readme.md [hunk 4]
```diff
@@ -80,244 +81,476 @@ As you're reviewing the app, try to fill in the blanks in the below Rails-to-Exp
...
+
+In `connection.js` we need to...
+  1. Remove any references to seed data from `connection.js`.
+  2. Set `module.exports = mongoose`.
+
+Now, create a new `db/seed.js` file. In it we will...
+  1. Require `connection.js` and `seeds.json`, saving them to their own `mongoose` and `seedData` variables respectively.  
+  2. Define a `Candidate` variable that will contain our Mongoose model definition.
+  3. Write Mongoose queries that accomplish the following...
+    - Clears the database of all data, and `.then`...
```
Comment (jshawl) [inline readme.md:172]: link to docs would be helpful for methods to use

File: readme.md [hunk 5]
```diff
@@ -80,244 +81,476 @@ As you're reviewing the app, try to fill in the blanks in the below Rails-to-Exp
...
+</details>
 
-> [f8df0b1](https://www.github.com/ga-wdi-exercises/whenpresident/commit/f8df0b1)
+<details>
+  <summary><strong>What does it mean to pass `{}` as an argument into `.remove()`?</strong></summary>
 
-- `connection.js` has `mongoose.model("Candidate", CandidateSchema)`. `seeds.js` has `mongoose.model("Candidate")`. What's the difference between these two statements? (Bonus: What's it similar to in Angular?)
-- What does it mean to pass `{}` as an argument into `.remove()`?
-- What does `process.exit()` do?
+  > Closes the file and prevents it from running any further.
```
Comment (jshawl) [inline readme.md]: `.remove({})` does this? i had no idea!

File: readme.md [hunk 6]
```diff
@@ -80,244 +81,476 @@ As you're reviewing the app, try to fill in the blanks in the below Rails-to-Exp
...
 
-### [Added seed data: `db/connection.js`](https://www.github.com/ga-wdi-exercises/whenpresident/blob/f8df0b1/db/connection.js)
-![Added seed data, db/connection.js](_DIFFSHOTS/added-seed-data.db-connection-js.png)
+  <br/>
+
+</details>
+
+![Add Seed Data to DB 1](http://i.imgur.com/zWHSpRO.png)
```
Comment (jshawl) [inline readme.md:206]: mind adding these images to the repo to prevent 404 in the future

File: readme.md [hunk 7]
```diff
@@ -80,244 +81,476 @@ As you're reviewing the app, try to fill in the blanks in the below Rails-to-Exp
...
+>  
 > **`.then(function(candidates){ ... })`** - `candidates` represents the all the Candidates pulled from the database. We can then reference this inside of `.then`.  
-
+>  
 > **`candidates: candidates`** - A little confusing, but the `candidates` we will be referencing in our view are now set to the `candidates` that are returned by Mongoose.  
 
-# Shows db candidate in candidates#show
+## You Do: Show (15 minutes / 1:30)
 
-> [05715e4](https://www.github.com/ga-wdi-exercises/whenpresident/commit/05715e4)
```
Comment (jshawl) [inline readme.md:169]: is there a commit students can check out if they fell behind?

File: readme.md [hunk 8]
```diff
@@ -80,244 +81,476 @@ As you're reviewing the app, try to fill in the blanks in the below Rails-to-Exp
...
+![Install body-parser](http://i.imgur.com/ZWSc01J.png)
+
+> **`var parser = require("body-parser")`** - Require `body-parser` so we can reference it later.  
+>  
 > **`app.use(parser.urlencoded({extended: true}))`** - configure the parser to support html forms
 
-# Added nonfunctional post route
+## You Do: New (10 minutes / 1:50)
 
-> [d596426](https://www.github.com/ga-wdi-exercises/whenpresident/commit/d596426)
```
Comment (jshawl) [inline readme.md:200]: is there another commit they can check out?

File: readme.md [hunk 9]
```diff
@@ -80,244 +81,476 @@ As you're reviewing the app, try to fill in the blanks in the below Rails-to-Exp
...
+* What does module.exports do?
+* Why does method="post" even though we are updating (vs. creating) something?
+* Why do we use promises and callbacks when calling methods on a Mongoose model?
 
-```bash
-$ git checkout express-mongoose-solution
-```
+## Homework
+
+You should now be able to complete the [second part of the YUM homework](https://github.com/ga-wdi-exercises/yum#part-ii-add-express).
```
Comment (jshawl) [inline readme.md:537]: have they done part 1?

File: readme.md (hunk 3 above)
Comment (amaseda) [inline readme.md]: Oops, yeah!

File: readme.md (hunk 9 above)
Comment (amaseda) [inline readme.md:537]: They were told that they learned enough to do Part I after Friday afternoon's class. With this class, they should now be able to do Part II. Whether they've actually done Part I at this point, I don't know.

File: readme.md (hunk 7 above)
Comment (amaseda) [inline readme.md:169]: My bad, meant to re-add these.

File: readme.md (hunk 4 above)
Comment (amaseda) [inline readme.md:172]: :+1:

**Pull request merged**